public/app/features/alerting/unified/components/receivers/form/fields/TemplateContentAndPreview.tsx

```diff
@@ -0,0 +1,93 @@
+import { css, cx } from '@emotion/css';
+import React from 'react';
+import AutoSizer from 'react-virtualized-auto-sizer';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Box, useStyles2 } from '@grafana/ui';
+import { useAlertmanager } from 'app/features/alerting/unified/state/AlertmanagerContext';
+import { GRAFANA_RULES_SOURCE_NAME } from 'app/features/alerting/unified/utils/datasource';
+
+import { EditorColumnHeader } from '../../../contact-points/templates/EditorColumnHeader';
+import { TemplateEditor } from '../../TemplateEditor';
+import { getPreviewResults } from '../../TemplatePreview';
+import { usePreviewTemplate } from '../../usePreviewTemplate';
+
+export function TemplateContentAndPreview({
+  payload,
+  templateContent,
+  templateName,
+  payloadFormatError,
+  setPayloadFormatError,
+  className,
+}: {
+  payload: string;
+  templateName: string;
+  payloadFormatError: string | null;
+  setPayloadFormatError: (value: React.SetStateAction<string | null>) => void;
+  className?: string;
+  templateContent: string;
+}) {
+  const styles = useStyles2(getStyles);
+
+  const { selectedAlertmanager } = useAlertmanager();
+  const isGrafanaAlertManager = selectedAlertmanager === GRAFANA_RULES_SOURCE_NAME;
+
+  const { data, error } = usePreviewTemplate(templateContent, templateName, payload, setPayloadFormatError);
+  const previewToRender = getPreviewResults(error, payloadFormatError, data);
+
+  return (
+    <div className={cx(styles.container, className)}>
+      <EditorColumnHeader label="Template content" />
+      <Box flex={1}>
+        <div className={styles.viewerContainer({ height: 400 })}>
+          <AutoSizer>
+            {({ width, height }) => (
+              <TemplateEditor
+                value={templateContent}
+                containerStyles={styles.editorContainer}
+                width={width}
+                height={height}
+                readOnly
+              />
+            )}
+          </AutoSizer>
+        </div>
+      </Box>
+
+      {isGrafanaAlertManager && (
+        <>
+          <EditorColumnHeader label="Preview with the default payload" />
+          <Box flex={1}>
+            <div className={styles.viewerContainer({ height: 300 })}>{previewToRender}</div>
+          </Box>
+        </>
+      )}
+    </div>
+  );
+}
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  container: css({
+    label: 'template-preview-container',
+    display: 'flex',
+    flexDirection: 'column',
+    borderRadius: theme.shape.radius.default,
+    border: `1px solid ${theme.colors.border.medium}`,
+  }),
+  viewerContainer: ({ height }: { height: number }) =>
+    css({
+      height,
```

Comment:
nit: should we turn this in to `minHeight`? We certainly have the vertical space for showing more and a large template (like the default one) will always overflow this container.

Reply:
But `AutoSizer` needs a fixed size in the parent, so for the template content we cannot use this `minHeight`

Reply:
`min-height` should allow the parent to have a height it can use for the auto-sizer – the editor won't grow beyond `400px` because once the `<AutoSizer>` has set the height the child component can no longer make the parent grow.

It _should_ allow the preview to grow though which is more likely to be larger than the template (because of iterations / loops).